camera_calibration/src/camera_calibration/calibrator.py

```diff
@@ -674,14 +697,27 @@ def set_alpha(self, a):
         original image are in calibrated image).
         """
 
-        # NOTE: Prior to Electric, this code was broken such that we never actually saved the new
-        # camera matrix. In effect, this enforced P = [K|0] for monocular cameras.
-        # TODO: Verify that OpenCV #1199 gets applied (improved GetOptimalNewCameraMatrix)
-        ncm, _ = cv2.getOptimalNewCameraMatrix(self.intrinsics, self.distortion, self.size, a)
-        for j in range(3):
-            for i in range(3):
-                self.P[j,i] = ncm[j, i]
-        self.mapx, self.mapy = cv2.initUndistortRectifyMap(self.intrinsics, self.distortion, self.R, ncm, self.size, cv2.CV_32FC1)
+        if "pinhole" in self.distortion_model:
+            # NOTE: Prior to Electric, this code was broken such that we never actually saved the new
+            # camera matrix. In effect, this enforced P = [K|0] for monocular cameras.
+            # TODO: Verify that OpenCV #1199 gets applied (improved GetOptimalNewCameraMatrix)
+            ncm, _ = cv2.getOptimalNewCameraMatrix(self.intrinsics, self.distortion, self.size, a)
+            for j in range(3):
+                for i in range(3):
+                    self.P[j,i] = ncm[j, i]
+            self.mapx, self.mapy = cv2.initUndistortRectifyMap(self.intrinsics, self.distortion, self.R, ncm, self.size, cv2.CV_32FC1)
+        elif "fisheye" in self.distortion_model:
+            # TODO: Implement re-computation of P given users balance value
```

Comment:
This TODO seems pretty important, can you clarify why this isnt implemented or what the plan is?

Reply:
For some reason `fisheye.estimateNewCameraMatrixForUndistortRectify` does not provide a meaningful camera matrix. It may be just an Opencv bug
If anybody knows how to use this function please let me know. It only misses the capability for the user to select the balance.
Similar issue: https://stackoverflow.com/questions/34316306/opencv-fisheye-calibration-cuts-too-much-of-the-resulting-image

Reply:
The first answer in that thread has what seems to be a real work around, is that not usable here for some reason?

Reply:
A naive workaround was implemented by scaling the projection matrix focal lengths in eb285f3b131975ed7465cc75e86d5d59a6880081